tests/components/climate/test_init.py

```diff
@@ -78,10 +79,24 @@ async def test_sync_turn_on(hass):
 
 async def test_sync_turn_off(hass):
     """Test if async turn_off calls sync turn_off."""
-    climate = MockClimateDevice()
+    climate = MockClimateEntity()
     climate.hass = hass
 
     climate.turn_off = MagicMock()
     await climate.async_turn_off()
 
     assert climate.turn_off.called
+
+
+def test_deprecated_base_class(caplog):
+    """Test deprecated base class."""
+
+    class CustomClimate(ClimateDevice):
+        def hvac_mode(self):
+            pass
+
+        def hvac_modes(self):
+            pass
+
+    CustomClimate()
+    assert "ClimateDevice is deprecated, modify CustomClimate" in caplog.text
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5418277)